### DISCUSSION TITLE
Change Font size and style of Field Lable and input

### DISCUSSION THREAD

**Discussion by Manishkparmar80** [Budibase/budibase#1879] [Help] [answered]
1. How i can Change Font size and style of Field Lable and input for individual field.

2. How i can Change Font size and style - Field Lable and input for entire application

**Comment by aptkingston** [comment, marked as answer]
Hey @Manishkparmar80. To change the size of form fields and labels, there's a setting on the `Form` component to pick between a medium and a large size. It's just below the Theme setting. This will change the size of all fields and field labels.
![image](https://user-images.githubusercontent.com/9075550/123916372-7c067400-d979-11eb-8e33-0420578c4124.png)

For font style, you can use custom CSS to change the font of field labels. You can't change the font style of the inputs themselves (and you probably shouldn't, we use an accessible font by default). To change the style of the labels, select either a field or the whole form and click "Custom CSS" and then enter whatever font style you want. For example:
![image](https://user-images.githubusercontent.com/9075550/123916938-1f578900-d97a-11eb-9f8c-21ae5fbc9622.png)

p.s. please don't use actually use Impact font, this is just an example 😄

**Comment by Manishkparmar80** [comment]
@aptkingston  thanks. i will try this out. 

In previous version,rthere was field size option to  reduce or increase width and height of field but new version many such design options has been removed. 

So how i can  reduce or increase width and height of field in new version.

**Reply by aptkingston** [reply]
The correct way to adjust the size of most components is to put them inside a `Container` or a `Section` and let those components take care of sizing.

You can however also use the "Custom CSS" option on the `Field Group` component that your fields are in. You can set `width: 400px;` or whatever width you want on your field group, and all the fields inside will use this width.

**Reply by arpit314ad** [reply]
After setting up the "Custom CSS" option on the Field Group, the above Field Group comes below and so on.
![Invoice](https://user-images.githubusercontent.com/86228170/124574336-ffd3cb00-de67-11eb-8290-951e8e45dfbe.JPG)

**Reply by aptkingston** [reply]
@arpit314ad horizontal forms aren't something the field group supports. If you want a horizontal form, you'd be better not using a field group at all. Your best bet would just be adding a container inside your form, setting the container as a row and then adding your fields inside your container and setting whatever custom widths you want on each form field directly.

**Comment by arpit314ad** [comment]
I tried with above suggestion but I am not able to get the  columns. Please suggest.
![Form01](https://user-images.githubusercontent.com/86228170/124591911-7ed1ff00-de7a-11eb-9878-537137d92982.JPG)

**Reply by aptkingston** [reply]
@arpit314ad sorry but I'm not sure what that screenshot is meant to show.
For a horizontal form, try nesting `Form` > `Container` > `Field`.
I made it quickly in an app to show you. I added a `width` custom CSS style to the options picker because it was the only one that needed it. Take a look at my component tree to get an idea of what I mean.
![image](https://user-images.githubusercontent.com/9075550/124599047-29d3c000-de5d-11eb-9bb6-319ea1fa7ed7.png)

**Comment by arpit314ad** [comment]
Hi...
Thanks for the revert but the issue is that how do I setup the data type from external data source??

**Reply by aptkingston** [reply]
Hi @arpit314ad. Please make a new discussion with some more details if you'd like help with a different issue. This one was about styling forms and field widths and is resolved.
